### ISSUE TITLE
Fix new TeamCity violations

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#4250]
https://teamcity.jetbrains.com/viewLog.html?buildId=1048700&tab=Inspection&buildTypeId=Checkstyle_IdeaInspectionsPullRequest

> Number of inspection errors 92 is 92 more than the provided threshold 0
Number of inspection warnings 192 is 192 more than the provided threshold 0

from logs:
```
[23:42:06] :	 [Step 2/2] Starting up IntelliJ IDEA 2017.1.2 (build IU-171.4249.21) ...done.
```
but latest available to download is 2017.1.1 . So this is just another experiment on TC with Intellij version.

We have repeatedly kept getting these errors in new PRs, so they must be valid and did another upgrade.

````
Code maturity issues
Deprecated member is still used (1)
 Code style issues
Replace Optional.isPresent() checks with functional-style expressions (3)
 Data flow issues
Scope of variable is too broad (Errors) (19)
 Declaration redundancy
Simplify stream API call chains (2)
 Dependency issues
Class with too many transitive dependents (Errors) (8)
 General
Annotator (2)
Redundant suppression (Errors) (2)
 Java 8
Anonymous type has shorter lambda alternative (2)
Lambda can be replaced with method reference (Errors) (1)
Loop can be collapsed with Stream API (Errors) (56)
Replace with Map.forEach (1)
Replace with single Map method (6)
 JUnit issues
Usage of obsolete 'junit.framework.Assert' method (Errors) (5)
 Naming conventions
Package naming convention (30)
 Probable bugs
Result of method call ignored (1)
Unused assignment (Errors) (1)
 RegExp
Duplicate branch in alternation (59)
Single character alternation (85)
````

Looking at a handfull, they seem to be valid and not false errors.

**Comment by Vladlis** [comment]
We might need to reconsider our TC config. I'm not sure that we need to fix this for example:
`Loop can be collapsed with Stream API (Errors) (56)`

**Comment by romani** [comment]
Let's disable these rules TC for now, they will be restored and fixed during GSoC project

**Comment by romani** [comment]
problem with rendering of 'Package naming convention (30)' is reported to TC support
https://teamcity-support.jetbrains.com/hc/en-us/requests/968250

**Comment by romani** [comment]
fix is merged.
All fixes were done on IntellijIdea 2017.1.1 .
